lib/types.nix

```diff
@@ -32,7 +39,22 @@ rec {
     check = v: (isRawType v) || (v ? __empty);
   };
 
-  maybeRaw = type: types.either type rawLua;
+  maybeRaw =
+    elemType:
+    let
+      luaFirst = types.either rawLua elemType;
+      elemFirst = types.either elemType rawLua;
+    in
+    luaFirst
+    // {
+      name = "maybeRaw";
+      inherit (elemFirst) description;
```

Comment:
Alternatively, we could construct a description ourselves. But due to nix's lazyness, calling `either` just to build a description is still fairly cheap.

If we wanted to do a custom description for maybeRaw, perhaps another PR can tackle that?

Reply:
For reference, here's `either`'s description [impl](https://github.com/NixOS/nixpkgs/blob/21be6da8c784dfd6284f39f9782960387a44ac62/lib/types.nix#L975-L981):
```nix
description =
  if t1.descriptionClass or null == "nonRestrictiveClause"
  then
    # Plain, but add comma
    "${t1.description}, or ${optionDescriptionPhrase (class: class == "noun" || class == "conjunction") t2}"
  else
    "${optionDescriptionPhrase (class: class == "noun" || class == "conjunction") t1} or ${optionDescriptionPhrase (class: class == "noun" || class == "conjunction" || class == "composite") t2}";
```